addons/stock/security/stock_security.xml

```diff
@@ -133,5 +133,20 @@
         <field name="global" eval="True"/>
         <field name="domain_force">['|',('company_id','=',False),('company_id','child_of',[user.company_id.id])]</field>
     </record>
+
+    <!-- Stock Portal Access Rules -->
+    <record id="stock_picking_rule_portal" model="ir.rule">
+        <field name="name">Portal Follower Pickings</field>
+        <field name="model_id" ref="stock.model_stock_picking"/>
+        <field name="domain_force">['|', '|', ('message_partner_ids', 'in', [user.partner_id.id]), ('partner_id.id', '=', user.partner_id.id), ('sale_id.partner_id.id', '=', user.partner_id.id)]</field>
```

Comment:
Should'tn we use something similar to tasks / issues, using child_of commecial partner id ? Probably to see with smart poeple like Odo, if necessary. Note this could be considered as an improvement and moved in another backlog.